src/builtins.c

```diff
@@ -677,10 +677,10 @@ JL_CALLABLE(jl_f_setfield)
         jl_type_error("setfield!", (jl_value_t*)jl_datatype_type, v);
     jl_datatype_t *st = (jl_datatype_t*)vt;
     if (!st->mutabl)
-        jl_errorf("type %s is immutable", jl_symbol_name(st->name->name));
+        jl_error("setfield! cannot change immutable struct");
```

Comment:
This seems strictly less informative than the previous text.

Reply:
Yes, but it's an error message—how much do we want to care?

Reply:
If we don't care then why change it at all?

Reply:
This isn't the text that you would usually see. That's text was emitted at https://github.com/JuliaLang/julia/pull/29721/files#diff-7a4c9160e84426f92c6790c4e8b14efaL2432

Reply:
> but it's an error message—how much do we want to care?

This is a terrible attitude. The quality of error messages is one of the primary things that makes people feel that a programming language is "helpful". For example, Elm gets rave reviews for the quality of the error messages. Similarly, in the past few years there's been a huge push to improve Rust's error messages with the very visible effect that what is still a fairly hard-to-learn language is considered to be much more usable than it once was.

If the other place where this message can be emitted from has a worse version of this message, then can we please change them both to be the better message instead of to the worse one?